common/cpp/src/google_smart_card_common/numeric_conversions.cc

```diff
@@ -32,10 +32,14 @@ bool CastDoubleToInt64(double value, int64_t* result,
                        std::string* error_message) {
   if (!(internal::kDoubleExactRangeMin <= value &&
         value <= internal::kDoubleExactRangeMax)) {
-    *error_message = FormatPrintfTemplate(
-        "The real value is outside the exact integer representation range: %f "
-        "not in [%" PRId64 "; %" PRId64 "]",
-        value, internal::kDoubleExactRangeMin, internal::kDoubleExactRangeMax);
+    if (error_message) {
+      *error_message = FormatPrintfTemplate(
+          "The real value is outside the exact integer representation range: "
+          "%f "
+          "not in [%" PRId64 "; %" PRId64 "]",
+          value, internal::kDoubleExactRangeMin,
+          internal::kDoubleExactRangeMax);
+    }
     return false;
   }
   *result = static_cast<int64_t>(value);
```

Comment:
In case you're wondering: yes, apparently I don't validate the double value to have the zero (or "almost zero") fractional part. This is orthogonal to the current WebAssembly effort, but probably we'll need to tighten this function to refuse fractional numbers.